tests/doc_test.go

```diff
@@ -0,0 +1,246 @@
+// +build system
+
+package tests
+
+/*
+Find valid ZQ examples in markdown, run them against
+https://github.com/brimsec/zq-sample-data, and compare results in docs with
+results produced.
+
+TODO:
+- Add some more tests
+- Parallelize
+- Use or update pkg/test for exec instead of my own
+
+Separate patches:
+- Deal with examples that need head
+- Unit tests for negative markdown cases
+- Find files as opposed to hard-coding them
+- bash -c sanitization
+*/
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/text"
+)
+
+type ZQExampleBlockType string
+
+/*
+Use these markers in markdown fenced code blocks to denote either a zq
+command or output from zq. Use is like:
+
+```zq-command
+zq "* | count()
+```
+```zq-output
+1234
+```
+
+This is in compliant with https://spec.commonmark.org/0.29/#example-113
+
+Doc authors MUST pair off each command and output in their own fenced code blocks.
+
+A zq-command code block MUST be one line after the leading line, which includes
+the info string.
+
+A zq-output fenced code block MAY be multiple lines. zq-output MUST be verbatim
+from actual zq output.
```

Comment:
@philrz While I work on some of @alfred-landrum's comments, could you let me know where you'd like to see these instructions? I have them as comments here for now. I'm happy to move them anywhere you'd like.

Note too that because I'm going to be getting rid of `bash -c` as the caller, there are going to be some updates to these instructions.

Reply:
I'm fine with them staying here for now.

BTW, as long as I'm writing a comment, a nit with some text above: compliant -> compliance. 😄